client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -136,25 +138,22 @@ private int getIdFromRegistry(String subject, Schema schema)
   @Override
   public synchronized int register(String subject, Schema schema)
       throws IOException, RestClientException {
-    Map<Schema, Integer> schemaIdMap;
-    if (schemaCache.containsKey(subject)) {
-      schemaIdMap = schemaCache.get(subject);
-    } else {
-      schemaIdMap = new IdentityHashMap<Schema, Integer>();
-      schemaCache.put(subject, schemaIdMap);
+    final Map<Schema, Integer> schemaIdMap =
+        schemaCache.computeIfAbsent(subject, k -> new IdentityHashMap<>());
```

Comment:
this method is already synchronized, so I'm assuming this is just to make it look more java8-y ?

Reply:
No threading issues. As per the description, this is just avoiding the double lookup pattern.

Reply:
@big-andy-coates we actually use `null` keys in some of the maps, that's why you had to use `containsKey` instead of `get`